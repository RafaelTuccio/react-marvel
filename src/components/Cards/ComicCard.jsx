const ComicCard = (props) => {
    let comic = {
        name: "spider man"
    }

    return(
        <div>
            <img src={comic} alt="quadrinhos de superherÃ³is" />
            <span><strong>{comic.name}</strong></span>
        </div>
    )
}

export default ComicCard;