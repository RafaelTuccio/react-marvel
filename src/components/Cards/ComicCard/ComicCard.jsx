const ComicCard = (props) => {
    let { comic } = props;
    let tamanho = "/portrait_incredible.";

    return(
        <div className="comic-card">
            <div className="comic-img">
                <img src={comic.thumbnail.path + tamanho + comic.thumbnail.extension} alt={comic.title} />    
            </div>
            <div className="comic-footer">
            <span><strong>{comic.title}</strong></span>
            </div>
        </div>
    );
}

export default ComicCard;