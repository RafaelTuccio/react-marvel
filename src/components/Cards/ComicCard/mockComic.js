export const comicMockUnica = {
    id: 89659,
    title: "Immortal Hulk (2018) #43",
    thumbnail: {
        path: "http://i.annihil.us/u/prod/marvel/i/mg/9/a0/6009e1e7e7ad7",
        extension: "jpg"
    },
    
}