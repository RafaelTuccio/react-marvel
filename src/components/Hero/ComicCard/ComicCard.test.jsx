import { render, screen } from '@testing-library/react';
import ComicCard from './ComicCard';
import { comicMockUnica } from './mockComic';


const comic = comicMockUnica;

describe("<ComicCard />", ()=> {
    it("deve renderizar normalmente na tela", () => {
        const {debug} = render(<ComicCard comic={comic}/>);
        debug()
        expect(screen.getByRole("img", {name: comic.title}));
        expect(screen.getByText("Immortal Hulk (2018) #43"));
        
    })

    it("deve dar match com o snapshot", () => {
        const { container } = render(<ComicCard comic={comic} />);
        expect(container.firstChild).toMatchSnapshot();
    })
    
})