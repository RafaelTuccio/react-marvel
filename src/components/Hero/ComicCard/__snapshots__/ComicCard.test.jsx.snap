// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicCard /> deve dar match com o snapshot 1`] = `
<div
  class="comic-card"
>
  <div
    class="comic-img"
  >
    <img
      alt="Immortal Hulk (2018) #43"
      src="http://i.annihil.us/u/prod/marvel/i/mg/9/a0/6009e1e7e7ad7/portrait_incredible.jpg"
    />
  </div>
  <div
    class="comic-footer"
  >
    <span>
      <strong>
        Immortal Hulk (2018) #43
      </strong>
    </span>
  </div>
</div>
`;
